src/cache/lru_cache_manager/lru_cache_manager.h

```diff
@@ -23,18 +23,18 @@ typedef struct LRUCacheManager
  * @param  capacity: Cache cacpcity
  * @retval Pointer to the cache manager
  */
-LRUCacheManager *LRUCacheManager_New(size_t capacity);
+LRUCacheManager *LRUCacheManager_New(size_t capacity, cacheValueFreeFunc freeCB);
 
 /**
  * @brief  Enqueue a new cache object in the cache
  * @note   Enqueuing is not an atomic procedure: 
  *         You first need to check if the cache is full, and if so, evict an entry
  * @param  *lruCacheManager: Cache manager address (pointer)
  * @param  *hashKey: cache entry key (charected arrau in size of HASH_KEY_LENGTH)
- * @param  *resultSet: cache entry value (ResultSet pointer)
+ * @param  *cacheValue: cache entry value 
  * @retval 
  */
-CacheData *addToCache(LRUCacheManager *lruCacheManager, const char *hashKey, ResultSet *resultSet);
+CacheData *addToCache(LRUCacheManager *lruCacheManager, unsigned long long const hashKey, void *cacheValue);
```

Comment:
Seems to me like this is over abstraction, my guess is that LRUQueue has an add function, which is what the cache needs to call directly.

Reply:
file removed